cpu/cc2538/periph/gpio.c

```diff
@@ -26,19 +26,27 @@
 #include "periph/gpio.h"
 
 #define GPIO_MASK           (0xfffff000)
-#define PORTNUM_MASK        (0x00007000)
+#define PORTNUM_MASK        (0x00003000)
 #define PORTNUM_SHIFT       (12U)
 #define PIN_MASK            (0x00000007)
 #define MODE_NOTSUP         (0xff)
 
 static inline cc2538_gpio_t *gpio(gpio_t pin)
 {
-    return (cc2538_gpio_t *)(pin & GPIO_MASK);
+    if(((uint32_t)pin &GPIO_MASK) == 0){
```

Comment:
thanks for clearing things up! So this `if ... else` is not _really_ needed (and should be fixed otherwise), and the actual fix is correction of `PORTNUM_MASK` and the interrupt initialization?